juriscraper/opinions/united_states/state/kan_p.py

```diff
@@ -1,113 +1,54 @@
 #  Scraper for Kansas Supreme Court
 # CourtID: kan_p
 
-from juriscraper.AbstractSite import logger
-from juriscraper.OpinionSiteLinearWebDriven import OpinionSiteLinearWebDriven
 
+from juriscraper.lib.html_utils import get_row_column_text
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
 
-class Site(OpinionSiteLinearWebDriven):
+
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.court = "Supreme Court"
         self.court_id = self.__module__
-        self.pages = []
         self.pages_to_process = 5
+        self.court = "Supreme Court"
         self.status = "Published"
         self.url = "https://www.kscourts.org/Cases-Opinions/Opinions.aspx"
 
-    def _download(self, request_dict={}):
-        if self.test_mode_enabled():
-            test_page = super()._download(request_dict)
-            self.pages.append(test_page)
-            return
-        self.initiate_webdriven_session()
-        self.click_through_filter_form()
-        self.paginate_through_search_results()
+    def _set_parameters(self, page: int) -> None:
+        """Set Parameters for Kansas Courts
+
+        :param page: Page number to request
+        :return: None
+        """
+        self.method = "POST"
+        self.parameters = {
+            "__EVENTARGUMENT": str(page),
+            "__EVENTTARGET": "p$lt$zonePagePlaceholder$pageplaceholder$p$lt$ctl06$UniversalPager$pagerElem",
+            "p$lt$zonePagePlaceholder$pageplaceholder$p$lt$ctl02$OpinionFilter1$filterControl$drpPublished": "Published",
+            "p$lt$zonePagePlaceholder$pageplaceholder$p$lt$ctl02$OpinionFilter1$filterControl$drpCourt": self.court,
+            "p$lt$zonePagePlaceholder$pageplaceholder$p$lt$ctl02$OpinionFilter1$filterControl$drpSortBy": "Opinion Date",
+        }
+        vs = self.html.xpath("//input[@id='__VIEWSTATE']")[0].get("value")
+        self.parameters["__VIEWSTATE"] = vs
 
     def _process_html(self):
-        path_non_header_rows = "//table[@class='info-table']//tr[position()>1]"
-        for page in self.pages:
-            for row in page.xpath(path_non_header_rows):
+        if self.test_mode_enabled():
+            self.pages_to_process = 1
+
+        for page in range(1, self.pages_to_process + 1):
+            if not self.test_mode_enabled():
+                self._set_parameters(page)
+                self.html = self._download()
```

Comment:
This is another place we should use `DeferringList` to make courts happy.

Reply:
Is the suggestion here that we might go to process pages that dont exist? I dont see how this pattern really works the way it does in ... say ND.

Reply:
No, the point is to not download a page unless we actually need it. Imagine two approaches. First yours:

1. We download each page in self.pages_to_process
2. We hand that off to CL.
3. CL starts downloading the related PDFs
4. After the fourth PDF, CL realizes there's nothing new and stops

Total downloads: self.pages_to_process + four PDFs.

If we use a DeferringList:

1. We download page 1.
2. We hand that to CL
3. CL starts downloading the related PDFs
4. After the fourth PDF, CL realizes there's nothing new, and stops

Total downloads: 1 + four PDFs.

Big picture, the idea is to avoid downloading pages we don't need.

Reply:
Can you look at the patten I just pushed @mlissner